resources/qml/Settings/SettingTextField.qml

```diff
@@ -98,9 +98,9 @@ SettingItem
 
             selectByMouse: true;
 
-            maximumLength: 10;
+            maximumLength: (definition.type == "[int]") ? 20 : 10;
 
-            validator: RegExpValidator { regExp: (definition.type == "int") ? /^-?[0-9]{0,10}$/ : /^-?[0-9]{0,9}[.,]?[0-9]{0,10}$/ } // definition.type property from parent loader used to disallow fractional number entry
+            validator: RegExpValidator { regExp: (definition.type == "[int]") ? /^\[(\s*-?[0-9]+\s*,)*(\s*-?[0-9]+)\s*\]$/ : (definition.type == "int") ? /^-?[0-9]{0,10}$/ : /^-?[0-9]{0,9}[.,]?[0-9]{0,10}$/ } // definition.type property from parent loader used to disallow fractional number entry
```

Comment:
The regex for integers is `-?[0-9]{0,10}`, as stated in the second option of the double ternary operator. The 0 in `{0,10}` is to allow removing the number altogether. It should actually be 1. The 10 should actually be infinite but we disallow numbers with more than 10 digits. If it were `{1,infinite}` that would be equivalent to `+`. So that becomes `-?[0-9]+`.

From this follows that the regex for comma separated lists of integers would be `(-?[0-9]+,)*(-?[0-9]+)`.

Allow for whitespace everywhere, and it becomes `(\s*-?[0-9]+\s*,)*(\s*-?[0-9]+)\s`.

Now enclose this in square brackets and it becomes `^\[(\s*-?[0-9]+\s*,)*(\s*-?[0-9]+)\s\]$`. This is exactly the regex that smartavionics got. The only difference I'd make is to limit the integer number length again.